src/bin/nydus-image/core/bootstrap.rs

```diff
@@ -540,9 +540,22 @@ impl Bootstrap {
 
         let blob_table_entries = blob_table.entries.len();
 
+        let (prefetch_table_offset, prefetch_table_size) = if ctx.prefetch.len() > 0 {
+            // Prefetch table is very close to blob devices table
+            let offset = blob_table_offset + blob_table_size;
+            let size = ctx.prefetch.len() * size_of::<u64>() as u32;
+            trace!("prefetch table locates at offset {} size {}", offset, size);
+            (offset, size)
+        } else {
+            (0, 0)
+        };
+
         let orig_meta_addr = bootstrap_ctx.nodes[0].offset;
         let meta_addr = if blob_table_size > 0 {
-            align_offset(blob_table_offset + blob_table_size, EROFS_BLOCK_SIZE as u64)
+            align_offset(
+                blob_table_offset + blob_table_size + prefetch_table_size as u64,
+                EROFS_BLOCK_SIZE as u64,
+            )
```

Comment:
What if `blob_table_size` is zero but `prefetch_table_size` is non-zero?

Reply:
It' impossible. If `blob_table_size` is zero, current image has no associated blob generated. Then prefetch table has nothing to prefetch. At least, the prefetch source comes from a blob.